test/jdk/sun/security/tools/jarsigner/AutoKeyStore.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8281234
+ * @summary The -protected option is not always checked in keytool and jarsigner
+ * @library /test/lib
+ * @modules java.base/sun.security.tools.keytool
+ *          java.base/sun.security.x509
+ */
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.SecurityTools;
+import jdk.test.lib.util.JarUtils;
+import sun.security.tools.keytool.CertAndKeyGen;
+import sun.security.x509.X500Name;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.*;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.jar.JarFile;
+
+public class AutoKeyStore {
+
+    public static void main(String[] args) throws Exception {
+
+        JarUtils.createJarFile(Path.of("unsigned.jar"), Path.of("."),
+                Files.writeString(Path.of("file"), "hello"));
+
+        SecurityTools.keytool("""
+                -J--add-exports -Jjava.base/sun.security.tools.keytool=ALL-UNNAMED
+                -J--add-exports -Jjava.base/sun.security.x509=ALL-UNNAMED
+                -providerClass AutoKeyStore$AutoProvider
+                -providerPath $test.classes
```

Comment:
You changed the code below due to missing "8281175: Add a -providerPath option to jarsigner", but here, there is still "-providerPath".

Reply:
The code above calls keytool which supports the argument. jarsigner does not. For jarsigner it was added by above mentioned change.